--- EntityID.java	2010-12-05 22:50:04.000000000 -0800
+++ newEntityID.java	2010-12-05 23:30:58.000000000 -0800
@@ -159,20 +159,37 @@
  } // end of unmarshal method 
 
 
- /**
-  * The equals method doesn't always work--mostly it works only on classes that consist only of primitives. Be careful.
-  */
- public boolean equals(EntityID rhs)
- {
-     boolean ivarsEqual = true;
-
-    if(rhs.getClass() != this.getClass())
-        return false;
-
-     if( ! (site == rhs.site)) ivarsEqual = false;
-     if( ! (application == rhs.application)) ivarsEqual = false;
-     if( ! (entity == rhs.entity)) ivarsEqual = false;
-
-    return ivarsEqual;
- }
+    /**
+     * This equals method actually does work. Go ahead and use it. It's useful
+     * for hash table keys when looking up entities.
+     */
+    public boolean equals(EntityID rhs)
+    {
+        boolean ivarsEqual = true;
+        
+        if(rhs.getClass() != this.getClass())
+            return false;
+        
+        if( ! (site == rhs.site)) ivarsEqual = false;
+        if( ! (application == rhs.application)) ivarsEqual = false;
+        if( ! (entity == rhs.entity)) ivarsEqual = false;
+        
+        return ivarsEqual;
+    }
+    
+    /**
+     * Override of base class hashcode. This is convienent to use when looking up entities in
+     * a hash table. It uses the application and entity IDs to fill out the 32 bits. The idea
+     * is to get a "pretty close" to unique hashcode for each entity ID in 32 bits, and the
+     * best way to do that is leave out the site ID while keeping the rest, which tends to
+     * be more unique.
+     */
+    public int hashcode()
+    {
+        int hashcode = site;
+        hashcode = hashcode << 16;
+        hashcode = hashcode + entity;
+        return hashcode;
+    }
+	
 } // end of class
