--- VariableDatum.java	2010-12-05 22:50:04.000000000 -0800
+++ newVariableDatum.java	2010-12-05 23:45:49.000000000 -0800
@@ -86,7 +86,7 @@
 @XmlAttribute
 @Basic
 public long getVariableDatumLength()
-{ return (long)variableDatums.size();
+{ return (long)variableDatums.size() * 64; // patch to fix units; bits rather than bytes
 }
 
 /** Note that setting this value will not change the marshalled value. The list whose length this describes is used for that purpose.
@@ -112,7 +112,7 @@
     try 
     {
        dos.writeInt( (int)variableDatumID);
-       dos.writeInt( (int)variableDatums.size());
+       dos.writeInt( (int)variableDatums.size() * 64 ); // post-processing patch to fix units; bits rather than bytes
 
        for(int idx = 0; idx < variableDatums.size(); idx++)
        {
@@ -132,6 +132,9 @@
     {
        variableDatumID = dis.readInt();
        variableDatumLength = dis.readInt();
+        int over = variableDatumLength % 64 > 0 ? 1 : 0; // post-processing patch to fix units problem
+        variableDatumLength = (variableDatumLength / 64) + over;
+        
        for(int idx = 0; idx < variableDatumLength; idx++)
        {
            EightByteChunk anX = new EightByteChunk();
